src/System.Security.Permissions/src/System/Security/Permissions/PrincipalPermission.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace System.Security.Permissions
+{
+    public sealed partial class PrincipalPermission : System.Security.IPermission, System.Security.ISecurityEncodable, System.Security.Permissions.IUnrestrictedPermission
+    {
+        public PrincipalPermission(System.Security.Permissions.PermissionState state) { }
+        public PrincipalPermission(string name, string role) { }
+        public PrincipalPermission(string name, string role, bool isAuthenticated) { }
+        public System.Security.IPermission Copy() { return default(System.Security.IPermission); }
+        public void Demand() { }
```

Comment:
This one permission type is different from the rest: it doesn't derive from CodeAccessPermission and it's not really about Code Access Security.  It's also something we might reasonably expect to see developers use as a way of restricting what user at the operating system level is running the app.  We should probably revisit all of PrincipalPermission to give it as real an implementation as we can muster.  For now, I'd suggest adding a TODO and making Demand throw a PlatformNotSupportedException.  (We'll likely want to build it on top of WindowsIdentity, but WindowsIdentity is currently only implemented for Windows, and while we will likely provide an implementation for at least some of it for Unix, we haven't yet.)